src/renderer/services/chain/memo.ts

```diff
@@ -0,0 +1,34 @@
+import { getDepositMemo } from '@thorchain/asgardex-util'
+import * as FP from 'fp-ts/lib/function'
+import * as O from 'fp-ts/lib/Option'
+import * as Rx from 'rxjs'
+import * as RxOp from 'rxjs/operators'
+
+import { BASE_CHAIN } from '../../const'
+import { eqChain } from '../../helpers/fp/eq'
+import { sequenceTOption } from '../../helpers/fpHelpers'
+import { selectedPoolAsset$ } from '../midgard/common'
+import { baseAddress$ } from './address'
+import { MemoRx } from './types'
+
+/**
+ * Stake memo
+ */
+const stakeMemo$: MemoRx = Rx.combineLatest([selectedPoolAsset$, baseAddress$]).pipe(
+  RxOp.switchMap(([oPoolAsset, oBaseAddress]) =>
+    Rx.of(
+      FP.pipe(
+        sequenceTOption(oPoolAsset, oBaseAddress),
```

Comment:
`sequenceTOption` will fall into `None`  value if one of element is `None`. im not sure here but what about scenario when we dont really need `oBaseAddress ` ? example
* oPoolAsset is asse from `BASE_CHAIN`
* oBaseAddress === O.none

in case of BASE_CHAIN we dont need on address anyway. 
do i miss anything ?

Reply:
Good questions + good point :+1: ! You are right, for deposits of base-chain assets, we don't need to add an address to memo. But in case of x-chain deposit we have to. Current implementation does not reflect all scenarios, I have fixed with 8cb51e2